engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1102,19 +1104,24 @@ && isAssignableFrom(argumentTypes[lastArgIndex], expressionTypes[lastArgIndex]))
                 }
             }
 
-            if (varArgType.isPrimitive() && allExpressionTypesArePrimitive) {
-                // there are ambiguous oddities with primitive varargs, so if its primitive lets just box it ourselves
+            if ((TypeUtils.isBoxedType(varArgType) || varArgType.isPrimitive()) && allExpressionTypesArePrimitive) {
```

Comment:
I'm a little confused about what's happening here. Does the complaint only apply if the inputs are primitive, which `allExpressionTypesArePrimitive` suggests we're limiting ourselves to, here? Shouldn't the primitive varargs be a better match, in that case?